.github/workflows/autotest.yml

```diff
@@ -5,7 +5,25 @@ on:
   - push
 
 jobs:
+  header-lint:
+    if: 'github.event_name == ''pull_request'''
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: |
+          git pull origin master
+          echo "Current git rev $(git rev-parse HEAD)"
+          echo "Git rev $(git merge-base HEAD remotes/origin/master)"
```

Comment:
Also, minor change, but let's rename this to 
```
echo "Common Ancestor git rev....
```

Reply:
Good idea! I'll change it as soon as we know how to deal with the other changes! 👍

Reply:
Also, another minor one,  we should add a `name` field to this step, otherwise, the runs will display the `git pull origin master` as their name :P 

So, something like
```
- name: Check headers
        run: |
          git pull origin master
```
should be good to go :D

Reply:
Changed